--- src/lib/support/Span.h	2024-12-10 20:34:34.000000000 +0800
+++ ../connectedhomeip-1.4.0.0-patched/src/lib/support/Span.h	2026-04-16 01:18:44.872539904 +0800
@@ -206,7 +206,7 @@
 
 inline namespace literals {
 
-inline constexpr Span<const char> operator"" _span(const char * literal, size_t size)
+inline constexpr Span<const char> operator""_span(const char * literal, size_t size)
 {
     return Span<const char>(Unchecked, literal, size);
 }
--- src/tools/chip-cert/chip-cert.cpp	2024-12-10 20:34:34.000000000 +0800
+++ ../connectedhomeip-1.4.0.0-patched/src/tools/chip-cert/chip-cert.cpp	2026-04-16 01:18:44.872851280 +0800
@@ -85,7 +85,7 @@
 
 } // namespace
 
-extern "C" int main(int argc, char * argv[])
+int main(int argc, char * argv[])
 {
     bool res = false;
 
