diff --git a/pw_hdlc/py/BUILD.gn b/pw_hdlc/py/BUILD.gn
--- a/pw_hdlc/py/BUILD.gn
+++ b/pw_hdlc/py/BUILD.gn
@@ -41,7 +41,6 @@
   python_test_deps = [
     "$dir_pw_build/py",
     "$dir_pw_log:protos.python",
-    "$dir_pw_tokenizer/py:test_proto.python",
   ]
   pylintrc = "$dir_pigweed/.pylintrc"
   mypy_ini = "$dir_pigweed/.mypy.ini"
diff --git a/pw_protobuf_compiler/proto.gni b/pw_protobuf_compiler/proto.gni
--- a/pw_protobuf_compiler/proto.gni
+++ b/pw_protobuf_compiler/proto.gni
@@ -80,7 +80,7 @@
     }
 
     _includes =
-        rebase_path(get_target_outputs(":${invoker.base_target}._includes"),
+        rebase_path(get_target_outputs(":${invoker.base_target}._includes_${current_toolchain}"),
                     root_build_dir)
 
     pw_python_action("$target_name._gen") {
@@ -111,7 +111,7 @@
       }
 
       deps = [
-        ":${invoker.base_target}._includes",
+        ":${invoker.base_target}._includes_${current_toolchain}",
         ":${invoker.base_target}._sources",
       ]
 
@@ -581,17 +581,17 @@
 
   # For each proto target, create a file which collects the base directories of
   # all of its dependencies to list as include paths to protoc.
-  generated_file("$target_name._includes") {
+  generated_file("$target_name._includes_${current_toolchain}") {
     # Collect metadata from the include path files of each dependency.
 
     deps = []
     foreach(dep, _deps) {
       _base = get_label_info(dep, "label_no_toolchain")
-      deps += [ "$_base._includes(" + get_label_info(dep, "toolchain") + ")" ]
+      deps += [ "$_base._includes_" + get_label_info(dep, "toolchain") + "(" + get_label_info(dep, "toolchain") + ")" ]
     }
 
     data_keys = [ "protoc_includes" ]
-    outputs = [ "${_common.base_out_dir}/includes.txt" ]
+    outputs = [ "${_common.base_out_dir}/includes-${current_toolchain}.txt" ]
 
     # Indicate this library's base directory for its dependents.
     metadata = {
diff --git a/pw_tokenizer/py/BUILD.gn b/pw_tokenizer/py/BUILD.gn
--- a/pw_tokenizer/py/BUILD.gn
+++ b/pw_tokenizer/py/BUILD.gn
@@ -56,7 +56,6 @@
     "tokens_test.py",
     "varint_test_data.py",
   ]
-  python_test_deps = [ ":test_proto.python" ]
   inputs = [
     "elf_reader_test_binary.elf",
     "example_binary_with_tokenized_strings.elf",
@@ -70,8 +69,3 @@
   ]
 }
 
-pw_proto_library("test_proto") {
-  sources = [ "detokenize_proto_test.proto" ]
-  deps = [ "..:proto" ]
-  prefix = "pw_tokenizer_tests"
-}
